### PR TITLE
GTEST/UCP: Fixed UCP AM test configuration.

### PR DESCRIPTION
**Pull request by rakhmets**
## What
Fixed reply configuration of test_ucp_am_nbx_seg_size.
Moved reply configuration to new parent class.

### CODE REVIEW

File: test/gtest/ucp/test_ucp_am.cc [hunk 1]
```diff
@@ -1525,11 +1525,6 @@ class test_ucp_am_nbx_rndv_dts : public test_ucp_am_nbx_rndv {
 
         test_ucp_am_nbx::cleanup();
     }
-
-    virtual unsigned enable_proto()
-    {
-        return 0;
-    }
 };
```
Comment (brminich) [inline test/gtest/ucp/test_ucp_am.cc:1528]: why was it incorrect?

Comment (rakhmets) [reply test/gtest/ucp/test_ucp_am.cc:1528]: Enabling/disabling new protocols should be done inside ctor.
Thus the method should be called inside ctor. Making it virtual may lead to not obvious behavior.

Comment (rakhmets) [reply test/gtest/ucp/test_ucp_am.cc]: These changes was merged within https://github.com/openucx/ucx/pull/8131

File: test/gtest/ucp/test_ucp_am.cc [hunk 2]
```diff
@@ -799,24 +799,34 @@ UCS_TEST_P(test_ucp_am_nbx, rx_am_mpools,
...
+public:
+    test_ucp_am_nbx_closed_ep()
+    {
+        modify_config("RESOLVE_REMOTE_EP_ID", "auto");
+    }
+
+    static void get_test_variants(std::vector<ucp_test_variant> &variants)
+    {
+        test_ucp_am_nbx_reply::get_test_variants(variants);
+    }
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_am.cc:829]: do we really have to define this? it's just going to superclass

File: test/gtest/ucp/test_ucp_am.cc [hunk 3]
```diff
@@ -965,9 +975,7 @@ class test_ucp_am_nbx_eager_data_release : public test_ucp_am_nbx {
 
     static void get_test_variants_reply(std::vector<ucp_test_variant> &variants)
     {
-        add_variant_values(variants, test_ucp_am_base::get_test_variants, 0);
-        add_variant_values(variants, test_ucp_am_base::get_test_variants,
-                           UCP_AM_SEND_FLAG_REPLY, "reply");
+        test_ucp_am_nbx_reply::get_test_variants(variants);
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_am.cc:978]: maybe remove get_test_variants_reply wrapper, and use test_ucp_am_nbx_reply::get_test_variants directly from  test_ucp_am_nbx_eager_data_release::get_test_variants

File: test/gtest/ucp/test_ucp_am.cc [hunk 4]
```diff
@@ -1062,14 +1065,7 @@ class test_ucp_am_nbx_align : public test_ucp_am_nbx {
...
     {
-        add_variant_values(variants, test_ucp_am_base::get_test_variants, 0);
-        add_variant_values(variants, test_ucp_am_base::get_test_variants,
-                           UCP_AM_SEND_FLAG_REPLY, "reply");
-    }
-
-    virtual unsigned get_send_flag()
-    {
-        return get_variant_value(1);
+        test_ucp_am_nbx_reply::get_test_variants(variants);
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_am.cc:1068]: can we remove get_test_variants wrapper?

File: test/gtest/ucp/test_ucp_am.cc [hunk 5]
```diff
@@ -1132,9 +1128,7 @@ class test_ucp_am_nbx_seg_size : public test_ucp_am_nbx {
 
     static void get_test_variants(std::vector<ucp_test_variant> &variants)
     {
-        add_variant_values(variants, test_ucp_am_base::get_test_variants, 0);
-        add_variant_values(variants, test_ucp_am_base::get_test_variants,
-                           UCP_AM_SEND_FLAG_REPLY, "reply");
+        test_ucp_am_nbx_reply::get_test_variants(variants);
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_am.cc:1131]: same here  - remove wrapper?

File: test/gtest/ucp/test_ucp_am.cc [hunk 6]
```diff
@@ -1179,15 +1172,20 @@ class test_ucp_am_nbx_dts : public test_ucp_am_nbx {
 
         ep_params.field_mask |= UCP_EP_PARAM_FIELD_ERR_HANDLING_MODE;
         ep_params.err_mode    = static_cast<ucp_err_handling_mode_t>(
-                                                          get_variant_value(3));
+                get_variant_value(4));
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_am.cc:1175]: maybe add helper function err_mode() { return get_variant_value(4); }

File: test/gtest/ucp/test_ucp_am.cc [hunk 7]
```diff
@@ -1179,15 +1172,20 @@ class test_ucp_am_nbx_dts : public test_ucp_am_nbx {
...
         return ep_params;
     }
 
-    static void get_test_dts(std::vector<ucp_test_variant>& variants)
+    static void get_test_dts_reply(std::vector<ucp_test_variant> &variants)
+    {
+        test_ucp_am_nbx_reply::get_test_variants(variants);
+    }
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_am.cc:1182]: IMO no need to add this wrapper

**Review by yosefe — DISMISSED** [review]

**Pull request merged**